framework/core/src/Database/DatabaseServiceProvider.php

```diff
@@ -22,7 +23,7 @@ class DatabaseServiceProvider extends AbstractServiceProvider
      */
     public function register()
     {
-        $this->container->singleton(Manager::class, function (Container $container) {
+        $this->container->singleton(Manager::class, function (ContainerImplementation $container) {
```

Comment:
We can't inject the interface here because of the custom `'flarum'` key?

Reply:
no, because we instantiate a laravel illuminate `Manager` class that expects the implementation rather than the contract. Another case of many where it's hard to use laravel's internals because it doesn't expect the contracts, we can use stubs to fix this as well if we want to.

Reply:
Let's keep it as implementation then; better to be accurate where we can/